dna/__main__.py

```diff
@@ -834,15 +886,18 @@ def agg_results_handler(arguments: argparse.Namespace):
     else:
         raise ValueError('no results to aggregate')
 
-    run_id = str(uuid.uuid4())
-    print(run_id)
+    results_to_agg = [json.load(open(results_path)) for results_path in tqdm(sorted(result_paths))]
+
+    results_ids = sorted([result['id'] for result in results_to_agg])
+    uuid_namespace = uuid.UUID('a8ec5977-701e-443b-89e7-8974e0e4ea6c')
+    run_id = str(uuid.uuid5(uuid_namespace, json.dumps(results_ids)))
+
     git_commit = utils.get_git_commit_hash()
 
     output_dir = os.path.join(getattr(arguments, 'output_dir'), run_id)
 
-    results_to_agg = [json.load(open(results_path)) for results_path in tqdm(sorted(result_paths))]
-
     agg_scores = aggregate_result_scores(results_to_agg)
+    plot.create_distribution_plots(agg_scores, output_dir)
```

Comment:
Instead of plotting in the `agg_results_handler`, I would like to have the plotting in the `report_handler`. This will allow us to plot the violin plots for all the models in the same plot. In the report handler, we loop over k (currently on line 654 in __main__.py) and you should be able to call your plotting function inside that loop. This will be a major change from the current plotting method. You can use `plot_at_k_scores_over_k` as a loose template for how to structure this new plotting method, with all models' violin plots in it.